src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -0,0 +1,356 @@
+                        rateStats.Delta.HasValue ? rateStats.Delta.Value.ToString() : "");
+                }
+                else if (stats.AggregationStatistics is LastValueStatistics lastValueStats)
+                {
+                    Logger.GaugeValuePublished(sessionId, instrument.Meter.Name, instrument.Meter.Version, instrument.Name, instrument.Unit, FormatTags(stats.Labels),
+                        lastValueStats.LastValue.HasValue ? lastValueStats.LastValue.Value.ToString() : "");
+                }
+                else if (stats.AggregationStatistics is HistogramStatistics histogramStats)
+                {
+                    Logger.HistogramValuePublished(sessionId, instrument.Meter.Name, instrument.Meter.Version, instrument.Name, instrument.Unit, FormatTags(stats.Labels), FormatQuantiles(histogramStats.Quantiles));
+                }
+            }
+
+            private string FormatTags(KeyValuePair<string, string>[] labels)
+            {
+                StringBuilder sb = new StringBuilder();
+                for (int i = 0; i < labels.Length; i++)
+                {
+                    sb.AppendFormat("{0}={1}", labels[i].Key, labels[i].Value);
+                    if (i != labels.Length - 1)
+                    {
+                        sb.Append(',');
+                    }
+                }
+                return sb.ToString();
+            }
+
+            private string FormatQuantiles(QuantileValue[] quantiles)
+            {
+                StringBuilder sb = new StringBuilder();
+                for (int i = 0; i < quantiles.Length; i++)
+                {
+                    sb.AppendFormat("{0}={1}", quantiles[i].Quantile.ToString(CultureInfo.InvariantCulture), quantiles[i].Value.ToString(CultureInfo.InvariantCulture));
+                    if (i != quantiles.Length - 1)
+                    {
+                        sb.Append(';');
+                    }
+                }
+                return sb.ToString();
+            }
+        }
+
+        private class MetricSpec
+        {
+            private static char s_meterInstrumentSeparator = '\\';
+            public string MeterName { get; private set; }
+            public string? InstrumentName { get; private set; }
+
+            public MetricSpec(string meterName, string? instrumentName)
+            {
+                MeterName = meterName;
+                InstrumentName = instrumentName;
+            }
+
+            public static bool TryParse(string text, out MetricSpec spec)
+            {
+                int slashIdx = text.IndexOf(s_meterInstrumentSeparator);
+                if (slashIdx == -1)
+                {
+                    spec = new MetricSpec(text.Trim(), null);
+                    return true;
+                }
+                else
+                {
+                    string meterName = text.Substring(0, slashIdx).Trim();
```

Comment:
>meterName [](http://example.com/codeflow?start=27&length=9)

if I pass empty string does it mean listening to all meters/instruments? so we only support specific meters and instruments? 

I am asking to know if we can listen to more than one meter and instrumts.

Reply:
Empty string would only match a meter whose name was empty string. It is possible to listen to multiple Meters/Instruments by passing a semicolon separated list as the Metrics argument. This test case shows an example: https://github.com/dotnet/runtime/pull/54333/files/f871425333a4f5a37e6a952f38f4943768db7d76#diff-8797eae6d3e644ed047f952e10d4240fe18175f754586666823b555f46968672R262

Reply:
do we support wild cards?

Reply:
We support using the name of the meter to include all instruments on that meter, but otherwise no. This comment is the full description of what is currently supported:
https://github.com/dotnet/runtime/pull/54333/files#diff-072334d2e8cacd4f335b7f48f0da8249a5619b6237e9ed311b029801145e8022R31-R34